wsnet/listen.go

```diff
@@ -316,3 +336,58 @@ func (l *listener) Close() error {
 func (l *listener) Addr() net.Addr {
 	return nil
 }
+
+// normalizeHost converts all representations of "localhost" to "localhost".
+func normalizeHost(addr string) string {
+	ip := net.ParseIP(addr)
+	if ip == nil {
+		return addr
+	}
+
+	if localNet.Contains(ip) {
+		return "localhost"
+	}
+	return addr
+}
+
+// getAddress parses the data channel's protocol into an address suitable for
+// net.Dial. It also verifies that the ProtoMessage permits connecting to said
+// address.
+func getAddress(msg ProtoMessage, protocol string) (netwk, addr string, err error) {
```

Comment:
Might make more sense to make this a function on `ProtoMessage`... something like:

```go
func (p ProtoMessage) Allowed(net, addr string) (bool, error) {

}
```

Then we can test it really well too... this is definitely a sensitive piece of code we wanna make sure is :bricks:

Reply:
Great idea!

Reply:
did this as `func (msg BrokerMessage) getAddress(protocol string) (netwk, addr string, err error)` instead